pkg/storage/stores/composite_store.go

```diff
@@ -73,9 +73,7 @@ func (c *CompositeStore) Stores() []Store {
 
 func (c compositeStore) Put(ctx context.Context, chunks []chunk.Chunk) error {
 	for _, chunk := range chunks {
-		err := c.forStores(ctx, chunk.From, chunk.Through, func(innerCtx context.Context, from, through model.Time, store Store) error {
-			return store.PutOne(innerCtx, from, through, chunk)
-		})
+		err := c.PutOne(ctx, chunk.From, chunk.Through, chunk)
```

Comment:
This seems unrelated to the purpose of the PR; some drive-by refactoring?
Wondering why we looped over all stores before.

Reply:
The [PutOne](https://github.com/grafana/loki/pull/7149/files/1cffed0ce83de39b74ab5d3873ae1031d3d98d9e#diff-099c12d62a2a013f805bf56cc21ce6af89456588cb905552b018def958512cb0R84) call below loops through all the stores. I just tried to reuse the code.

Reply:
Ah! I see, thanks.